CUJExecutor/app/src/androidTest/java/com/example/firstapp/ShellUtility.java

```diff
@@ -483,32 +512,35 @@ public String zeroPad(String s, int finalLength) {
         return sBuilder.toString();
     }
 
+
     public void logData(long[][] allActionStamps) {
```

Comment:
Instead of a 2d array, consider using an array of an object to represent your timestamps.